src/command-helpers/scaffold.js

```diff
@@ -106,18 +113,33 @@ const writeSchema = async (abi, protocol, schemaPath, entities) => {
 
 const writeMapping = async (abi, protocol, contractName, entities) => {
   const events = protocol.hasEvents()
-    ? abiEvents(abi).filter(event => entities.indexOf(event.get('name')) === -1).toJS()
+    ? abiEvents(abi)
+        .filter(event => entities.indexOf(event.get('name')) === -1)
+        .toJS()
     : []
 
-  let mapping = prettier.format(
-    generateEventIndexingHandlers(
-        events,
-        contractName,
-      ),
-    { parser: 'typescript', semi: false },
-  )
+  let mapping = prettier.format(generateEventIndexingHandlers(events, contractName), {
+    parser: 'typescript',
+    semi: false,
+  })
+
+  await fs.writeFile(`./src/${strings.kebabCase(contractName)}.ts`, mapping, {
+    encoding: 'utf-8',
+  })
+}
+
+const writeTestsFiles = async (abi, contractName) => {
```

Comment:
It is not part of codegen

Reply:
Also the utils files are not necessarily static, because the mocked events have default values, which someone may want to change, for example the `address`

Reply:
Ok. If it's before codegen, it's still referencing files from the codegen step (e.g. `import { ExampleEvent } from "../generated/schema"`) . Seems like you would want to regenerate the utils file every time `codegen` is run no?

Reply:
This is the entity the `init` command will generate for you. In the test case you are referencing, the entity is called `ExampleEvent`, because it tests when the `indexEvents` option is set to `true`. In this case `init` will generate entities for each event with the same name as the event. And to answer and your other question, those are example files with example tests that are meant to match the mapping generated by init. You'll probably have to update them yourself to match your logic. Those are meant to lay a foundation and get you started.